### PR TITLE
[MSBuildHelpers, MSBuildV1, VSBuildV1, XamarinAndroidV1] Enable fix for MSBuild.Logger

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
**Module names:** `MSBuildHelpers` ; `MSBuildV1` ; `VSBuildV1` ; `XamarinAndroidV1`

**Description:** This Pull Request enables the fix for `MSBuild.Logger` used by `msbuild.exe` during the `MSBuild` task execution.

**Documentation changes required:** `N`

**Added unit tests:** `N`

**Modified unit tests:** `Y`

**[Attached related issue](https://github.com/microsoft/build-task-team/issues/1176)**

**Checklist:**
- [x] `MSBuildHelpers` and tasks versions were bumped - [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected — built and deployed (using tfx) MSBuild task to the ADO organization.
    The task works as expected.

**ChangeLog:**

* **common-npm-packages/MSBuildHelpers**
    * `InvokeFunctions.ps1` — new input `|enableOrphanedProjectsLogs=true` added for `MSBuild.Logger`
    * `make.json` — archive url updated
    * `package.json` and `package-lock.json` — MSBuildHelpers version bumped up

* **Tasks**
    * **Common/MSBuildHelpers**
        * `InvokeFunctions.ps1` — new input `|enableOrphanedProjectsLogs=true` added for `MSBuild.Logger`
        * `make.json` — archive url updated
        * `package.json` and `package-lock.json` — MSBuildHelpers version bumped up
    * **MSBuildV1**
        * `task.json` and `task.loc.json` — task version bumped up
        * `package.json` and `package-lock.json` — MSBuildHelpers version bumped up
    * **VSBuildV1**
        * `task.json` and `task.loc.json` — task version bumped up
    * **XamarinAndroidV1**
        * `task.json` and `task.loc.json` — task version bumped up

* **Tasks/Common/MSBuildHelpers/Tests** +
  **common-npm-packages/MSBuildHelpers/Tests** (`L0.ts` and 5 unit tests for each of them)
    * `L0.ts` — timeout changed from 20 to 30 seconds (20 is not always enough)
    * New input `|enableOrphanedProjectsLogs=true` added for `MSBuild.Logger`
        * `Invoke-MSBuild.CombinesMsbuildexe.ps1`
        * `Invoke-MSBuild.FailsRootTimelineDetailOnExitCode.ps1`
        * `Invoke-MSBuild.OmitsTimelineDetail.ps1`
        * `Invoke-MSBuild.SkipsLogUploadIfMissing.ps1`
        * `Invoke-MSBuild.WritesRootTimelineDetail.ps1`

### CODE REVIEW

File: common-npm-packages/MSBuildHelpers/InvokeFunctions.ps1 [hunk 1]
```diff
@@ -108,7 +108,7 @@ function Invoke-MSBuild {
             # Hook up the custom logger.
             $loggerAssembly = "$PSScriptRoot\msbuildlogger\Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll"
             Assert-VstsPath -LiteralPath $loggerAssembly -PathType Leaf
-            $arguments = "$arguments /dl:CentralLogger,`"$loggerAssembly`";`"RootDetailId=$($detailId)|SolutionDir=$($solutionDirectory)`"*ForwardingLogger,`"$loggerAssembly`""
+            $arguments = "$arguments /dl:CentralLogger,`"$loggerAssembly`";`"RootDetailId=$($detailId)|SolutionDir=$($solutionDirectory)|enableOrphanedProjectsLogs=true`"*ForwardingLogger,`"$loggerAssembly`""
```
Comment (anatolybolshakov) [inline common-npm-packages/MSBuildHelpers/InvokeFunctions.ps1:111]: Should the common package version be bumped as well, and for tasks which import this - also?

Comment (DenisRumyantsev) [reply common-npm-packages/MSBuildHelpers/InvokeFunctions.ps1:111]: There are two places which cause concern:
* the version has **_never_** been bumped up here (perhaps we can update it to `1.196.0`)
https://github.com/microsoft/azure-pipelines-tasks/blob/9a99462589ea2816fea28afbed111dd31297244f/common-npm-packages/package.json#L3
* the version has been bumped up here a couple of times (perhaps we can update it to `2.196.0`)
https://github.com/microsoft/azure-pipelines-tasks/blob/9a99462589ea2816fea28afbed111dd31297244f/common-npm-packages/MSBuildHelpers/package.json#L3

Comment (anatolybolshakov) [reply common-npm-packages/MSBuildHelpers/InvokeFunctions.ps1:111]: We don't need to update the first - please bump the second. Using of sprint number make sense definitely - for better tracking.

Comment (DenisRumyantsev) [reply common-npm-packages/MSBuildHelpers/InvokeFunctions.ps1:111]: Done!

**Comment by anatolybolshakov** [comment]
@DenisRumyantsev will this change be picked up by XamarinAndroid and VSBuild tasks? Have you tested this for these tasks as well?

**Comment by DenisRumyantsev** [comment]
@anatolybolshakov yes, tested. These changes will be picked up by XamarinAndroid and VSBuild tasks

**Review by anatolybolshakov — APPROVED** [review]

**Review by alexander-smolyakov — APPROVED** [review]

**Comment by takost** [comment]
LGTM! Changed tasks work as expected

**Review by takost — APPROVED** [review]

**Review by tatyana-kostromskaya — APPROVED** [review]

**Comment by tatyana-kostromskaya** [comment]
LGTM! Changed tasks work as expected

**Pull request merged**